hashintel/hash
# Add entity archival endpoint

## 🌟 What is the purpose of this PR?



This PR adds an endpoint for archiving an entity.
This also allowed entity updates to be re-enabled

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1203250001255275/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?




- Fixes entity updating (and makes use of historic table)
- Adds new DB methods for moving an entity from `entities` to `entity_histories`
  - Optionally marks the entity as archived
- Adds HTTP POST endpoint for archiving
- Adds HTTP manual test for archiving
- Re-enables HTTP manual test for updates

## 📜 Does this require a change to the docs?




## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



-

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Run manual HTTP tests
1.  Confirm that the database has 2 entities in the latest table, `entities`, and 3 entity versions in the `entity_histories` table (1 for the update, 2 for the archival.)

## 📹 Demo
